app/routes.py

```diff
@@ -390,3 +394,21 @@ def forums():
 @app.route('/profile')
 def profile():
     return render_template('profile.html', title='CONP | My Profile', user=current_user)
+
+@app.route('/pipeline-search', methods=['GET'])
+def pipeline_search():
+    if request.method == 'GET':
+        import time
+        from app.threads import UpdatePipelineData
+        if not os.path.isfile(".cache/boutiques/pipelineData.json"):
+            if not os.path.exists(".cache/boutiques"):
+                os.makedirs(".cache/boutiques")
+            thr = UpdatePipelineData()
+            thr.start()
+            thr.join()
+        elif time.time() - os.path.getmtime(".cache/boutiques/pipelineData.json") > 300:
+            UpdatePipelineData().start()
+
+        #search cache and return result
+
+        return json.dumps("{}")
```

Comment:
You could dump the whole content of `pipelineData.json` in case no search query is provided, as is the case in `bosh search`.